MovieApp/DataLayer/Clients/BaseApiClient.swift

```diff
@@ -0,0 +1,72 @@
+import Foundation
+
+class BaseApiClient {
+
+    private let baseUrl: String
+    let apiKey = "e24dd8d2f3822e3917d10c6570d7f574"
+
+    init(baseUrl: String) {
+        self.baseUrl = baseUrl
+    }
+
+    func get<T: Decodable>(
+        path: String,
+        queryParameters: [String: String?]? = nil,
+        completionHandler: @escaping (Result<T, RequestError>) -> Void
+    ) {
+        var urlString = "\(baseUrl)\(path)"
+
+        if let parameters = queryParameters {
+            var parameterString = "?"
+            for (index, (key, value)) in parameters.enumerated() {
+                parameterString += "\(key)=\(value ?? "")"
+                if index != parameters.count-1 {
+                    parameterString.append("&")
+                }
+            }
+            urlString.append(parameterString)
+        }
+
+        guard let url = URL(string: urlString) else {
+            completionHandler(.failure(.invalidURLError))
+            return
+        }
+
+        let urlRequest = URLRequest(url: url)
```

Comment:
Since this method is a get, I would like that you explicitly set it in the URL request class.